glib/src/source.rs

```diff
@@ -872,14 +872,14 @@ where
 /// The default main loop almost always is the main loop of the main thread.
 /// Thus, the closure is called on the main thread.
 #[doc(alias = "g_unix_fd_add_full")]
-pub fn unix_fd_add<F>(fd: RawFd, condition: IOCondition, func: F) -> SourceId
+pub fn unix_fd_add<F>(fd: impl AsFd, condition: IOCondition, func: F) -> SourceId
```

Comment:
unsafe versions of these too

Reply:
(callback still uses `BorrowedFd`)

Reply:
I don't know how we can do better regarding the callback to be honest. Cant have 

```
F: FnMut(impl AsFd, IOCondition) -> ControlFlow + Send + 'static,
```

Reply:
No I meant for the callback it's fine like it is